features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetricDispatcher.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.domain.scope.RumViewManagerScope
+import com.datadog.android.rum.model.ViewEvent
+import java.util.concurrent.ConcurrentHashMap
+
+internal class SessionEndedMetricDispatcher(private val internalLogger: InternalLogger) :
+    SessionMetricDispatcher {
+
+    private val metricsBySessionId = ConcurrentHashMap<String, SessionEndedMetric>()
+
+    override fun startMetric(
+        sessionId: String,
+        startReason: RumSessionScope.StartReason
+    ) {
+        metricsBySessionId[sessionId] = SessionEndedMetric(sessionId, startReason)
+    }
+
+    override fun endMetric(sessionId: String) {
+        // the argument is always non - null, so we can suppress the warning
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        val metric = metricsBySessionId.remove(sessionId)
```

Comment:
Actually it's already declared here:
https://github.com/DataDog/dd-sdk-android/blob/develop/detekt_custom.yml#L204
Since it's a java function, it can throw NPE if the parameter is null, we just need to suppress it since in kotlin we are sure that the param is not null

Reply:
This declaration is a bit strange, because what should through NPE is the following one (notice the nullable type as an argument):

```
"java.util.concurrent.ConcurrentHashMap.remove(kotlin.String?):java.lang.NullPointerException"
```

while `"java.util.concurrent.ConcurrentHashMap.remove(kotlin.String)` cannot throw (because of non-nullable type).

Reply:
I fixed in detekt_custom.yml

Reply:
I don't see any change here or in `detekt_custom.yml` regarding that. Probably you didn't push a commit?